checkstyle/suppressions.xml

```diff
@@ -272,6 +272,9 @@
     <suppress checks="MethodCount|NPathComplexity|MagicNumber" files="com/hazelcast/spi/Operation"/>
     <suppress checks="MethodCount" files="com/hazelcast/spi/impl/NodeEngineImpl"/>
     <suppress checks="ClassFanOutComplexity" files="com/hazelcast/spi/impl/NodeEngineImpl"/>
+    <!-- node engine constructs objects, so it knows many of them -->
+    <suppress checks="ClassDataAbstractionCoupling" files="com/hazelcast/spi/impl/NodeEngineImpl"/>
```

Comment:
I'm in favor of the annotation style as well. Another advantage: If you refactor a class you directly see the suppressions and can give it a try to remove them. A suppression in the XML is easily forgotten, even if the class is improved.

But I won't make this a blocker if we can't get a consensus on this. Would be an easy task to convert most of the XML config to annoations in a separate PR.

I also made a suggestion to get rid of this specific suppression for `NodeEngineImpl` (no suppression at all should be the best solution anyway).